modules/tensor_mechanics/src/auxkernels/RankTwoScalarAux.C

```diff
@@ -52,8 +53,34 @@ RankTwoScalarAux::computeValue()
    case 6:
      val = calcEigenValues();
      break;
+   case 7: // VolumetricStrain (Lagrangian)
+     val = _tensor[_qp].trace() +
+           _tensor[_qp](x,x)*_tensor[_qp](y,y) +
+           _tensor[_qp](y,y)*_tensor[_qp](z,z) +
+           _tensor[_qp](z,z)*_tensor[_qp](x,x) +
+           _tensor[_qp](x,x)*_tensor[_qp](y,y)*_tensor[_qp](z,z);
+     break;
+   case 8: // FirstInvariant
+     val = _tensor[_qp].trace();
+     break;
+   case 9: // SecondInvariant
```

Comment:
I'm not sure what is going on in RankTwoTensor, but there is a problem with
my version.  (Assumes symmetry).  I'll push a new version.

On Tue, Oct 27, 2015 at 11:07 AM, John W. Peterson <notifications@github.com
> wrote:

> In modules/tensor_mechanics/src/auxkernels/RankTwoScalarAux.C
> <https://github.com/idaholab/moose/pull/5874#discussion_r43152557>:
>
> > @@ -52,8 +53,34 @@ RankTwoScalarAux::computeValue()
> >     case 6:
> >       val = calcEigenValues();
> >       break;
> > +   case 7: // VolumetricStrain (Lagrangian)
> > +     val = _tensor[_qp].trace() +
> > +           _tensor[_qp](x,x)*_tensor[_qp](y,y) +
> > +           _tensor[_qp](y,y)*_tensor[_qp](z,z) +
> > +           _tensor[_qp](z,z)*_tensor[_qp](x,x) +
> > +           _tensor[_qp](x,x)*_tensor[_qp](y,y)*_tensor[_qp](z,z);
> > +     break;
> > +   case 8: // FirstInvariant
> > +     val = _tensor[_qp].trace();
> > +     break;
> > +   case 9: // SecondInvariant
>
> Perhaps this is the sum of principal minors form
> <https://en.wikipedia.org/wiki/Invariants_of_tensors>?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/5874/files#r43152557>.
>



-- 
Jason D. Hales, PhD
Department Manager
Fuels Modeling & Simulation
Idaho National Laboratory
(208) 526-2293